integrations/slack/src/misc/utils.ts

```diff
@@ -290,50 +293,76 @@ export const getSyncState = async (client: Client, ctx: IntegrationCtx): Promise
   return payload as SyncState
 }
 
-export const getOAuthAccessToken = async (client: Client, ctx: IntegrationCtx) => {
-  const { state } = await client
-    .getState({ type: 'integration', name: 'credentials', id: ctx.integrationId })
-    .catch(() => ({
-      state: { payload: {} as any },
-    }))
+const getCredentials = async (client: Client, ctx: IntegrationCtx) => {
+  const {
+    state: { payload: credentials },
+  } = await client.getState({ type: 'integration', name: 'credentials', id: ctx.integrationId })
 
-  return state.payload.accessToken
+  return credentials
 }
 
 export const getAccessToken = async (client: Client, ctx: IntegrationCtx) => {
-  if (ctx.configuration.botToken) {
+  if (ctx.configurationType === 'botToken') {
     return ctx.configuration.botToken
   }
 
-  return getOAuthAccessToken(client, ctx)
+  return (await getCredentials(client, ctx)).accessToken
 }
 
-export const validateRequestSignature = ({ req, logger }: { req: Request; logger: IntegrationLogger }): boolean => {
-  const signingSecret = bp.secrets.SIGNING_SECRET
+export const getSigningSecret = async (client: Client, ctx: IntegrationCtx) => {
+  if (ctx.configurationType === 'botToken') {
+    return (await getCredentials(client, ctx)).signingSecret
+  }
+  return bp.secrets.SIGNING_SECRET
+}
 
-  const timestamp = req.headers['x-slack-request-timestamp']
-  const slackSignature = req.headers['x-slack-signature'] as string
+export class SlackEventSignatureValidator {
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement. --> 
**🤓 nitpick:** I prefer using functions instead of class methods when a class method can be a static function with arguments. Classes are good for managing internal states, but pure functions are easier for input/output.

This is a nitpick; you don't have to change anything.

Reply:
The signature validation code used to be a single, somewhat long function with a cyclomatic complexity of 10. I've broken it down into smaller chunks that are easier to reason about. The class serves as a logical grouping for these closely related functions. I could just keep these functions outside of a class, but using a class allows us to leverage private methods, keeping implementation details hidden and providing a clear public interface.